be/src/vec/exec/volap_scan_node.cpp

```diff
@@ -221,7 +221,14 @@ void VOlapScanNode::scanner_thread(VOlapScanner* scanner) {
             std::lock_guard<std::mutex> l(_free_blocks_lock);
             _free_blocks.emplace_back(block);
         } else {
-            blocks.push_back(block);
+            if (!blocks.empty() && blocks.back()->rows() + block->rows() <= _runtime_state->batch_size()) {
```

Comment:
Do we need a threshold? For example, if the full block size is 1024, but each coming block size is 1023, than we have to  merge block everytime, and there is even a `mutex` here

Reply:
`blocks.back()->rows() + block->rows() <= _runtime_state->batch_size() `
1023 + 1023 > 1024
so do not do merge here